server/controllers/recipe-controller.js

```diff
@@ -47,11 +47,75 @@ class RecipeController {
         preparationTime
       })
         .then(({ dataValues }) => {
-          successResponse(res, { recipe: dataValues }, 201);
+          return successResponse(res, { recipe: dataValues }, 201);
         })
         .catch(err => errorResponse(res, err, 400));
     } catch (error) {
-      validationErrorResponse(res, error.details);
+      return validationErrorResponse(res, error.details);
+    }
+  }
+
+  /**
+   * Update Recipe
+   *
+   * @static
+   * @param {*} req
+   * @param {*} res
+   * @param {Funtion} next
+   * @memberof RecipeController
+   * @returns {undefined}
+   */
+  static async updateRecipe(req, res) {
+    const { id } = req.user;
+    const { slug } = req.params;
+    const { body } = req;
+    const { title, ingrident, steps, cookingTime, preparationTime } = req.body;
+    try {
+      await validate(body, recipeUpdateSchema);
+      const recipe = await Recipe.findOne({ where: { slug, userId: id } });
+      if (!recipe) {
+        return errorResponse(res, 'recipe not found');
+      }
+      const data = await recipe.update(
+        {
+          title: title || recipe.title,
+          ingrident: ingrident || recipe.ingredient,
+          cookingTime: cookingTime || recipe.cookingTime,
+          preparationTime: preparationTime || recipe.preparationTime,
+          steps: steps || recipe.steps
+        },
+        { returning: true }
+      );
+      successResponse(res, { message: 'update successful', recipe: data }, 200);
+    } catch (error) {
+      errorResponse(res, error.message);
+    }
+  }
+
+  /**
+   * Delete Recipe
+   *
+   * @static
+   * @param {*} req
+   * @param {*} res
+   * @param {Funtion} next
+   * @memberof RecipeController
+   * @returns {undefined}
+   */
+  static async deleteRecipe(req, res) {
+    const { slug } = req.params;
+    try {
+      const data = await Recipe.findOne({ where: { slug } });
+      if (!data) {
+        return errorResponse(res, 'data not found ', 400);
```

Comment:
Wouldn't this be a 404? Also same up above with the update endpoint

Reply:
noted

Reply:
noted

Reply:
done